internal/metrics/metrics.go

```diff
@@ -187,6 +227,172 @@ func GetScalingSignal(pdb *policyv1.PodDisruptionBudget) string {
+	if isHelpful {
+		helpfulLabel = LikelyHelpfulStr
+	}
+	
+	return pattern, helpfulLabel
+}
+
+// AnalyzePodAgeFailurePattern analyzes the failure pattern of pods based on their age
+func AnalyzePodAgeFailurePattern(ctx context.Context, c client.Client, pdb *policyv1.PodDisruptionBudget) (string, error) {
+	// Convert PDB label selector to Kubernetes selector
+	selector, err := metav1.LabelSelectorAsSelector(pdb.Spec.Selector)
+	if err != nil {
+		return AllHealthyPattern, err
+	}
+
+	// List all pods matching the PDB selector
+	var podList corev1.PodList
+	err = c.List(ctx, &podList, &client.ListOptions{
+		Namespace:     pdb.Namespace,
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return AllHealthyPattern, err
+	}
+
+	if len(podList.Items) == 0 {
+		return AllHealthyPattern, nil
+	}
+
+	// Sort pods by creation time 
+	sort.Slice(podList.Items, func(i, j int) bool {
+		return podList.Items[i].CreationTimestamp.Before(&podList.Items[j].CreationTimestamp)
+	})
+
+	// Categorize pods as unhealthy
+	var unhealthyPods []corev1.Pod
+	for _, pod := range podList.Items {
+		if !isPodHealthy(&pod) {
+			unhealthyPods = append(unhealthyPods, pod)
+		}
+	}
+
+	// If all pods are healthy
+	if len(unhealthyPods) == 0 {
+		return AllHealthyPattern, nil
+	}
+
+	// Analyze failure pattern 
+	totalPods := len(podList.Items)
+	totalUnhealthyPods := len(unhealthyPods)
+	
+	// If less than 2 pods, can't determine pattern clearly
+	if totalPods < 2 {
+		return AllHealthyPattern, nil
+	}
+
+	// Split pods into oldest half and newest half
+	oldestHalf := totalPods / 2
+	
+	oldestFailureCount := 0
+	newestFailureCount := 0
+
+	for i, pod := range podList.Items {
+		if !isPodHealthy(&pod) {
+			if i < oldestHalf {
+				oldestFailureCount++
+			} else {
+				newestFailureCount++
+			}
+		}
+	}
+
+	// Follow majority logic
+	// If more than 66% of failures are in oldest half -> oldest failing
+	// If more than 66% of failures are in newest half -> newest failing  
+	// Otherwise this is a random failing pattern
+	oldestFailureRatio := float64(oldestFailureCount) / float64(totalUnhealthyPods)
+	newestFailureRatio := float64(newestFailureCount) / float64(totalUnhealthyPods)
+	
+	if oldestFailureRatio >= 0.66 {
+		return OldestFailingPattern, nil
+	} else if newestFailureRatio >= 0.66 {
+		return NewestFailingPattern, nil
+	} else {
+		return RandomFailingPattern, nil
+	}
+}
+
+// isPodHealthy determines if a pod is healthy based on its conditions
+func isPodHealthy(pod *corev1.Pod) bool {
+	// Check if pod is running
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// Check readiness condition
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady {
+			return condition.Status == corev1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+// PredictScalingEffectiveness predicts whether scaling up would be helpful based on failure pattern
+func PredictScalingEffectiveness(pattern string) bool {
```

Comment:
methods that aren't used outside package should be lower case. 
Be cautious about what you expose its your defactor api. Default to lower case and only expose what you need